### ISSUE TITLE
Subsetting while cells have an identity value of NA excludes those cells

### ISSUE THREAD

**Issue by jeremycfd** [satijalab/seurat#1070]
I've noticed that if you are subsetting a Seurat object (identifying the cells using cells.use and their respective cell barcodes) if the Seurat cell "identity" for any cell is set to an NA value that cell will not be included in the output subset. It took me a while to figure this out because it doesn't really make sense that the identity of a cell, which I feel most would consider metadata, would be used to exclude a cell when trying to subset based on cell identifiers that are included in the actual data. I assume this is a bug, but in any case if others are having the same issue I've found that just setting the identity of all cells to something allows you to get around it.

Thanks!

**Comment by timoast** [comment]
Hi, thanks for reporting this. It does appear to be a bug, I was able to reproduce this in Seurat v2 and Seurat v3.

Seurat v2:

```
> pbmc_small@ident[[5]]
[1] 0
Levels: 0 1 2 3
> cells.use <- WhichCells(pbmc_small, ident = 0)
> pbmc_small@ident[[5]] <- NA
> cluster0 <- SubsetData(pbmc_small, cells.use = cells.use)
> cells.use[[5]] %in% cluster0@cell.names
[1] FALSE
```

Seurat v3:

```
> Idents(pbmc_small)[[5]]
[1] 0
Levels: 0 1 2
> cells.use <- WhichCells(pbmc_small, idents = 0)
> Idents(pbmc_small)[[5]] <- NA
> cluster0 <- subset(pbmc_small, cells = cells.use)
> cells.use[[5]] %in% colnames(cluster0)
[1] FALSE
```

**Comment by andrewwbutler** [comment]
This should be fixed now in v3 as of 6c7e5a6.
